src/extensions/common-api/proxy.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { asLegacyGlobalFunctionForExtensionApi } from "../as-legacy-globals-for-extension-api/as-legacy-global-function-for-extension-api";
+import { resolveProxyInjectionToken } from "../../behaviours/proxy/resolve-proxy/common/resolve-proxy-injection-token";
+
+export const resolveProxy = asLegacyGlobalFunctionForExtensionApi(resolveProxyInjectionToken);
```

Comment:
Is there a "non-legacy" way to export functions for extensions?

Reply:
`resolveProxy` as a name is very generic (would `resolveSystemProxy` make more sense?). What kind of behavior extension api user expects from this? 🤔 At least we should have api docs (tsdoc) that shortly summarizes what it is.

Reply:
> Is there a "non-legacy" way to export functions for extensions?

All stuff exposed in current extension-API is involuntary **global state**. A completely clean function can be exported as it is, "non-legacy"... But that other stuff with side-effects and lifecycles and dependencies and all, cannot be but legacy :/

This is a considerable damage that could be fluently solved in next iteration of extension-API.

Reply:
> would resolveSystemProxy make more sense?

A bit I guess. Renamed :)

Reply:
TSDoc added from conversation elsewhere.